src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblyTools.cs

```diff
@@ -0,0 +1,155 @@
+        {
+            foreach (Type type in types)
+            {
+                TypeBuilder tb = mb.DefineType(type.FullName, type.Attributes, type.BaseType);
+
+                MethodInfo[] methods = type.IsInterface ? type.GetMethods() : type.GetMethods(BindingFlags.DeclaredOnly);
+                foreach (var method in methods)
+                {
+                    MethodBuilder meb = tb.DefineMethod(method.Name, method.Attributes, method.CallingConvention, method.ReturnType, null);
+                }
+
+                foreach (FieldInfo field in type.GetFields())
+                {
+                    tb.DefineField(field.Name, field.FieldType, field.Attributes);
+                }
+            }
+        }
+
+        internal static void WriteAssemblyToStream(AssemblyName assemblyName, Type[] types, Stream stream)
+        {
+            WriteAssemblyToStream(assemblyName, types, stream, null);
+        }
+
+        internal static void WriteAssemblyToStream(AssemblyName assemblyName, Type[] types, Stream stream, List<CustomAttributeBuilder>? assemblyAttributes)
+        {
+            MethodInfo defineDynamicAssemblyMethod = PopulateMethods(typeof(Stream), out MethodInfo saveMethod);
+
+            AssemblyBuilder assemblyBuilder = (AssemblyBuilder)defineDynamicAssemblyMethod.Invoke(null,
+                new object[] { assemblyName, CoreMetadataAssemblyResolver.s_coreAssembly, assemblyAttributes });
+
+            ModuleBuilder mb = assemblyBuilder.DefineDynamicModule(assemblyName.Name);
+
+            PopulateMembersForModule(types, mb);
+
+            saveMethod.Invoke(assemblyBuilder, new object[] { stream });
+        }
+
+        internal static MethodInfo PopulateMethods(Type parameterType, out MethodInfo saveMethod)
+        {
+            Type assemblyType = Type.GetType(
+                    "System.Reflection.Emit.AssemblyBuilderImpl, System.Reflection.Emit",
+                    throwOnError: true)!;
+
+            saveMethod = assemblyType.GetMethod("Save", BindingFlags.NonPublic | BindingFlags.Instance, new Type[] { parameterType });
+
+            return assemblyType.GetMethod("DefinePersistedAssembly", BindingFlags.NonPublic | BindingFlags.Static,
+                new Type[] { typeof(AssemblyName), typeof(Assembly), typeof(List<CustomAttributeBuilder>) });
+        }
+
+        internal static Assembly TryLoadAssembly(string filePath)
+        {
+            // Get the array of runtime assemblies.
+            string[] runtimeAssemblies = Directory.GetFiles(RuntimeEnvironment.GetRuntimeDirectory(), "*.dll");
+
+            // Create the list of assembly paths consisting of runtime assemblies and the inspected assembly.
+            var paths = new List<string>(runtimeAssemblies);
+            paths.Add(filePath);
+
+            // Create PathAssemblyResolver that can resolve assemblies using the created list.
+            var resolver = new PathAssemblyResolver(paths);
```

Comment:
```suggestion
            var resolver = new CoreMetadataAssemblyResolver();
```
Can we just use this - to match the Stream case? I do not see a good reason why we want to use a more complicated resolver here.

Reply:
I thought for loading from path needs PathAssemblyResolver 😛